gpii/node_modules/flowManager/src/FlowManager.js

```diff
@@ -141,6 +150,94 @@ fluid.defaults("gpii.flowManager.local", {
+    gradeNames: ["fluid.component"],
+    events: {
+        onPrepareCapture: null,
+        onPrepareSolutions: null,
+        onSolutions: null,
+        onReadyToCapture: {
+            events: {
+                prepareCapture: "onPrepareCapture",
+                prepareSolutions: "onPrepareSolutions",
+                solutions: "onSolutions"
+            },
+            args: [
+                "{arguments}.solutions.0",
+                "{arguments}.prepareCapture.0"
+            ]
+        },
+        onError: null
+    },
+    listeners: {
+        "onPrepareCapture.deviceContext": {
+            funcName: "gpii.flowManager.getDeviceContext",
+            args: ["{flowManager}.deviceReporterDataSource", "{that}.events.onPrepareSolutions"]
+        },
+        onPrepareSolutions: {
+            func: "{that}.getSolutions",
+            args: ["{arguments}.0"]
+        },
+        onReadyToCapture: {
+            func: "{that}.captureSystemSettings",
+            args: ["{arguments}.0", "{arguments}.1"]
+        },
+        onError: {
+            funcName: "console.log",
+            args: ["{arguments}.0"]
+        }
+    },
+    invokers: {
+        getSolutions: {
+            funcName: "gpii.flowManager.getSolutions",
+            args: [ "{flowManager}.solutionsRegistryDataSource", "{arguments}.0", "{that}.events.onSolutions", "{that}.events.onError"] //"{request}.events.onError"]
+        },
+        captureSystemSettings: {
+            funcName: "gpii.flowManager.captureSystemSettings",
+            args: ["{lifecycleManager}.invokeSettingsHandlerGet", "{lifecycleManager}.variableResolver", "{arguments}.0", "{arguments}.1"]
+        }
+    }
+});
+
+
+/**
+ * Runs through all the solutions currently available on the system, pulls the current
+ * setting for each supportedSetting and returns them in an object. Primary use case
+ * is for backing Capture tools that would allow a user to set up their GPII profile
+ * starting with the current settings for their applications on the local machine.
+ *
+ * @param {lifecycleManager.variableResolver} resolver - Resolver for looking up handler specifications
+ * @param {Object} solutions - Solutions registry entries for solutions available on the current machine.
+ * @param {event} event - Event to call with the payload as an argument after processing.
+ * @return Returns nothing. Results are fired as an argument to the `event` argument.
+ */
+gpii.flowManager.captureSystemSettings = function (invokeSettingsHandlerGet, resolver, solutions, event) {
+    var handlerGetPromises = [];
+    fluid.each(solutions, function (solution, solutionID) {
+        var newSolution = gpii.lifecycleManager.transformSolutionSettings(solution, gpii.settingsHandlers.changesToSettings);
+        fluid.each(newSolution.settingsHandlers, function (handlerSpec) {
+            //if applicationSettings is unspecified, dont attempt to get settings
+            if (!handlerSpec.supportedSettings) {
```

Comment:
I assume this is transitional, since there may be solutions which only support common terms, etc.